src/server/_query.py

```diff
@@ -375,7 +410,19 @@ def __str__(self):
         group_by = f"GROUP BY {_join_l(self.group_by)}" if self.group_by else ""
         index = f"USE INDEX ({self.index})" if self.index else ""
 
-        return f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} {group_by} {order}"
+        # if no signal array, assemble the sql and return
+        if not self.signal_array:
+            command = f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} {group_by}" 
+        else:
+            # if there is a signal array, concatenate signals with UNION ALL
+            command = f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} AND {self.signal_array[0]} {group_by}"
+            i = 1
+            while i < len(self.signal_array):
```

Comment:
is `i` strictly necessary here? why not use something like

```
for signal_clause in self.signal_array[1:]:
```

?

Reply:
or even something like 

```
"UNION ALL".join(
    f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} AND {signal_clause} {group_by}"
    for signal_clause in self.signal_array
)
```

Reply:
Yeah, I used the second one